### PR TITLE
Create loop-arbitrary-sequential-steps.yaml. Fixes #8081 

### LINKED ISSUE #8081: Examples should be added for failfast & withParam
# Summary
Before all, I am new to all K8S and argo workflow stuff.
Recently I am struggling with finding solutions for having sequential task pipeline of arbitrary steps. After endless searching on official argo documentation and googling, I finally ended up at: https://github.com/argoproj/argo-workflows/issues/3644. And that is exactly what I need, failfast & withParam!  
I strongly recommend that we have a example for that simple but splendid solution in repos. If authors are currently too busy, I am willing to write it myself.

# Use Cases

For those new to argo workflow.

---
**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by MagicFrogSJTU**
Docs examples: Add loop-arbitrary-sequential-steps example.

Fixes #8081

### CODE REVIEW

**Review by terrytangyuan — CHANGES_REQUESTED** [review]
Thanks. Please fix codegen.

File: examples/loop-arbitrary-sequential-steps.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+metadata:
+  generateName: loop-arbitrary-sequential-steps-
+spec:
+  entrypoint: loop-arbitrary-sequential-steps-example
+  arguments:
+    parameters:
+    - name: step_params
+      value: |
+        [
+          { "exit_code": 0, "message": "succ 1" },
```
Comment (simster7) [inline examples/loop-arbitrary-sequential-steps.yaml:16]: Minor: I think this is clearer
```suggestion
          { "exit_code": 0, "message": "succeeds 1" },
```

**Review by simster7 — CHANGES_REQUESTED** [review]

**Comment by MagicFrogSJTU** [comment]
I will fix all these within today

**Comment by MagicFrogSJTU** [comment]
Sorry. Been quite occupied these days. I will fix this stuff within this weekend.

File: examples/loop-arbitrary-sequential-steps.yaml (hunk 1 above)
Comment (MagicFrogSJTU) [reply examples/loop-arbitrary-sequential-steps.yaml]: Sorry the forced push ruined the requested change. Is there anything else to fix?

**Comment by MagicFrogSJTU** [comment]
> Thanks. Please fix codegen.

Sorry I don't understand the CodeGen error. Does it mean the added workflow should exit with code 0 when run?

**Comment by MagicFrogSJTU** [comment]
Sorry. Been quite occupied these days. I will fix this stuff within this weekend.

**Comment by alexec** [comment]
`codegen` is fixed by running `make pre-commit -B`

**Pull request closed without merging**